extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php

```diff
@@ -44,15 +48,45 @@ private function getFilePrefixRegex( $contLang ) {
 	 * @return mixed
 	 */
 	public function sanitizeImageFileName( $filename, $contLang ) {
-		// replace the MW square brackets and surrounding whitespace
-		$trimmedFilename = trim( $filename, "\t\n\r[]" );
-
+		$plainText = $this->convertToPlainText( $filename );
 		$filePrefixRegex = $this->getFilePrefixRegex( $contLang );
+		$textLines = explode( PHP_EOL, $plainText );
+
+		foreach ( $textLines as $potentialFilename ) {
+			$filename = $this->extractFilename( $potentialFilename, $filePrefixRegex );
+			if ($filename) {
+				return $filename;
+			}
+
+		}
+		return $plainText;
+	}
+
+	/**
+	 * @param $filename
+	 * @return string
+	 */
+	private function convertToPlainText( $filename ) {
+		// strip HTML tags
+		$filename = strip_tags( $filename );
+		// replace the surrounding whitespace
+		$filename = trim( $filename, "\t\n\r" );
```

Comment:
is there a reason for stating those white chars explicitly? please refer to http://php.net/trim